### PR TITLE
fix(issue_alerts): Cap maximum number of issue alerts per project to 100 (WOR-1843)

### PR DESCRIPTION
**Pull request by wedamija**
To prevent potential abuse use cases where users could potentially take down sentry we want to add a
limit of 100 rules per project.

### CODE REVIEW

File: src/sentry/api/endpoints/project_rules.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 from sentry.signals import alert_rule_created
 from sentry.web.decorators import transaction_start
 
+MAX_RULES_PER_PROJECT = 100
```
Comment (AniketDas-Tekky) [inline src/sentry/api/endpoints/project_rules.py:23]: Can we make this configurable for Self hosted?

Comment (wedamija) [reply src/sentry/api/endpoints/project_rules.py:23]: I had that thought too, will do

**Review by barkbarkimashark — APPROVED** [review]
Looks reasonable enough for me.

Do we surface up the error in the UI if the user goes over the limit? Main concern is the 400 response is a silent error.

**Comment by wedamija** [comment]
> Looks reasonable enough for me.
> 
> Do we surface up the error in the UI if the user goes over the limit? Main concern is the 400 response is a silent error.

Yeah, lower down if validation fails we also return a 400 with the errors from the serializer:
```
return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
The ui will surface those.

**Pull request merged**